trieste/models/gpflow/models.py

```diff
@@ -206,6 +228,7 @@ def optimize(self, dataset: Dataset) -> None:
             )
 
         self.optimizer.optimize(self.model, dataset)
+        self._posterior = self.model.posterior()  # update cached posterior
```

Comment:
I *believe* it isn't necessary to recreate this object, and you should be able to do `self._posterior.update_cache()` or the same function with some other enum. Would need to read the docs properly though.